CIME/SystemTests/test_utils/user_nl_utils.py

```diff
@@ -25,8 +25,12 @@ def append_to_user_nl_files(caseroot, component, contents):
             matching the pattern 'user_nl_clm*'. (We do a wildcard match to
             handle multi-instance tests.)
 
-        contents (str): Contents to append to the end of each user_nl file
+        contents (str or list): Contents to append to the end of each user_nl
+            file. If list, each item will be appended on its own line.
     """
+    
+    if not isinstance(contents, list):
```

Comment:
I think it would be more robust to change this to `if isinstance(contents, str)`: this way, tuples and other list-like objects will be handled the same as lists.

Reply:
Changed to `if isinstance(contents, str)` in commit 1e31efc.
Unit test added in commit ab8e713.